swiftui-project/LoginView.swift

```diff
@@ -0,0 +1,142 @@
+//  Created by user238749 on 5/6/23.
+//
+
+import SwiftUI
+
+struct LoginView: View {
+    @State private var email = "";
+    @State private var password = "";
+    @State private var isLoginValid = false;
+    @State private var isPasswordVisible = false;
+    @State private var isEmailValid = true;
+    @State private var showModal = false;
+    @State private var path: [String] = [];
+    
+    var body: some View {
+        NavigationStack(path: $path){
+            VStack{
+                Image("Logo")
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .frame(width: 250, height: 160, alignment: .center)
+                    .padding(.top, 50)
+                    .padding(.bottom,50)
+                
+                Text("Log in")
+                    .font(.largeTitle)
+                    .fontWeight(.bold)
+                    .foregroundColor(Color(UIColor(named: "TextColor")!))
+                    .frame(maxWidth:.infinity, alignment: .leading)
+                    .padding(.leading, 30)
+                
+                TextField("Email", text: $email)
+                    .autocapitalization(.none)
+                    .padding()
+                    .background(Color.white)
+                    .cornerRadius(10.0)
+                    .padding(.horizontal,26)
+                    .padding(.bottom,20)
+                    .onChange(of: email) {value in
+                        self.isEmailValid = validateEmail(value)
+                    }
+                if !isEmailValid {
+                    Text("Invalid email")
+                        .foregroundColor(.red)
+                    
+                }
+                HStack {
+                    if isPasswordVisible{
+                        TextField("Password", text: $password)
+                            .autocapitalization(.none)
+                            .padding()
+                            .background(Color.white)
+                            .foregroundColor(isLoginValid ? .black : .red)
+                        
+                    } else {
+                        SecureField("Password", text: $password)
+                            .autocapitalization(.none)
+                            .padding()
+                            .background(Color.white)
+                    }
+                    Button(action: {
+                        self.isPasswordVisible.toggle()
+                    }) {
+                        Image(systemName: isPasswordVisible ? "eye.slash" : "eye")
+                            .foregroundColor(.gray)
+                    }
+                }
+                .background(Color.white)
+                .cornerRadius(10.0)
+                .padding(.horizontal, 26)
+               
+               
+                    Button(action: {
+                             self.validateLogin()
+                        if isLoginValid {
+                            path.append("text")
```

Comment:
Here NavigationLink instead of path will be more appropriate

Reply:
Changed.